plugins/inputs/procstat/procstat.go

```diff
@@ -323,6 +323,10 @@ func (p *Procstat) updateProcesses(pids []PID, tags map[string]string, prevInfo
 			if name, _ := info.Name(); name == "" {
 				continue
 			}
+			if exists, err := info.Exists(); err != nil || !exists {
+				// process has gone
+				continue
+			}
```

Comment:
This does filter out pids that are no longer running. I'm wondering if returning non-running pids is really a problem though. I assume the pids we filter out here were running when the original query was run. In other words if we query pids, then one or more of them finishes before we get here, why should we drop them?

Are you using the data in a way that needs the pids to be running or is this intended to drop pids that have been finished for a long time?

Reply:
In the exported data they are shown as 'alive'. Yes, we need the information that the process is gone.

Reply:
The list of processes produced by the plugin is always going to be a little out of date because it's a snapshot in time. When the sample interval happens, the plugin checks which processes are running. Adding the .Exists() check here only modifies the snapshot.

I don't think it's valuable to double check with .Exists(). The snapshot was accurate when it was taken.

There are other use cases that don't want non-running pids to be removed like this. If we accept this change I suspect it will cause trouble for other users.